typhon/utils.py

```diff
@@ -31,6 +32,57 @@
 ui_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'ui')
 
 
+def UsesEnums(*enum_classes, base=QWidget):
+    '''
+    Add Qt Designer compatible enums to a class. Under the hood, this generates
+    a base class for the target class.
+
+    Parameters
+    ----------
+    *enum_classes : enum.Enum or namespace
+        Enum classes - can be either of type `enum.Enum` or a simple namespace
+        where one attribute is associated with one value.
+    base : class
+        The base class for the generated Qt-compatible class. Must be either
+        QObject or QWidget (or have one in the class hierarchy)
+    '''
+    class EnumBase(base):
+        for enum in enum_classes:
+            Q_ENUMS(enum)
+
+        _qt_enums = enum_classes
```

Comment:
What happens here if I inherit from a class that also inherits from `UseEnums`? Are we going to overwrite one?

Reply:
You answered your above question with this one. `_qt_enums` gets wiped away, hence the need for the discovery function above.

Thinking a bit more about it, I suppose this could be refactored with an `__init_subclass__`-style thing to keep the list up-to-date, though that would pin the discovery functionality to 3.6+ of course.